app/src/main/java/com/rasteplads/festfriend/FestFriendApplication.kt

```diff
@@ -19,13 +33,46 @@ import com.rasteplads.festfriend.pages.MapPage
 import com.rasteplads.festfriend.ui.theme.FestFriendTheme
 import kotlinx.coroutines.delay
 
+enum class JoinOrCreate { Join, Create, None }
+
 @Composable
 fun FestFriendApplication(appViewModel: AppViewModel = viewModel()){
 
     val appState by appViewModel.uiState.collectAsState()
     val navController = rememberNavController()
-    val navToMap = {navController.navigate(FestFriendScreen.Map.name)}
-    FestFriendTheme (dynamicColor = false){
+    val navToMap = { navController.navigate(FestFriendScreen.Map.name) }
+    val context = LocalContext.current
+    val locationClient = remember {
+        LocationServices.getFusedLocationProviderClient(context)
+    }
+    var action = JoinOrCreate.None
+    val permission = Manifest.permission.ACCESS_FINE_LOCATION
+    val locationPermissionLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted ->
+        if (isGranted) {
+            if (action == JoinOrCreate.Join)
+                appViewModel.joinGroup(navToMap)
+            else if (action == JoinOrCreate.Create)
+                appViewModel.createGroup(navToMap)
+        } else {
+            val builder = AlertDialog.Builder(context)
+            builder.setMessage("This app requires your location to work as expected")
```

Comment:
Måske lav en function og kald den her, så det er lidt tydligere at det er en error message

Reply:
i stedet for builder.builder.something.something
så bare AlertMessage(context, title, msg) eller sådan noget